### PR TITLE
docs: Add contribution guidelines

### LINKED ISSUE #464: Improving contribution guidelines
`CONTRIBUTING.md` is currently very out of date and needs to be revised.
We should link to this file form the main README.
The document should include:
- notes on our testing procedure
- step-by-step instructions on how to PR changes
- notes about `.md` files in stories
- ???

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #464

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

### CODE REVIEW

File: docs/COMMIT-GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+Examples:
+
+```
+See: https://github.com/xxx/yyy/
+See: 49d89b4acebd80838303b011d30517cd6229fdbe
+Link: https://github.com/xxx/yyy/issues/zzz
+```
+
+### `Closes: <url>`/`Fixes: <url>`
```
Comment (dimitrisnl) [inline docs/COMMIT-GUIDELINES.md]: Why not `Connects-To`?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+- Write a descriptive pull request title.
+
+- Squash commits when possible, for example, when committing review changes.
+
+Before your pull request can be merged, the following conditions must hold:
+
+- The linter doesn't throw any warning.
+
+- All the tests passes.
```
Comment (dimitrisnl) [inline docs/CONTRIBUTING.md]: Nit: `pass`

File: docs/COMMIT-GUIDELINES.md (hunk 1 above)
Comment (LucianBuzzo) [reply docs/COMMIT-GUIDELINES.md]: Using `Fixes` will automatically close the attached issue on merge, which is ideal for rendition as the package is autodeployed after every merge.

Comment (dimitrisnl) [reply docs/COMMIT-GUIDELINES.md]: Oh that's great to know 👍

**Pull request merged**